### DISCUSSION TITLE
Show plugins with update in printer list

### DISCUSSION THREAD

**Discussion by taker218** [OctoFarm/OctoFarm#873] [Implemented]
Hello,

I'm using OctoFarm for a while now and it would be nice to see, which plugins need to be updated before I'm updating them via OctoFarm.
Right now I'm going to the webinterface of my printers and see which ones are due to update.

It would be nice to see the plugin-names in the confirmation dialogue when updating multiple printers.

**Comment by NotExpectedYet** [comment]
> Hello,
> 
> I'm using OctoFarm for a while now and it would be nice to see, which plugins need to be updated before I'm updating them via OctoFarm. Right now I'm going to the webinterface of my printers and see which ones are due to update.
> 
> It would be nice to see the plugin-names in the confirmation dialogue when updating multiple printers.

As far as I remember it shows which plugins are to be updated... might have been another screen it's been a while since I've looked at the master version. 

I will post some screenshots of V1.2 shortly regardless as this has been improved.

**Reply by taker218** [reply]
You can see the pending updates when you click on the update button on the printer itself, not on the button to update all printers with updates :)

**Reply by NotExpectedYet** [reply]
> You can see the pending updates when you click on the update button on the printer itself, not on the button to update all printers with updates :)

Ah makes sense! Yeah that's currently the same on V1.2 as well. 

I left it like that because I was unsure what a good work around would be for plugin updates not been available on all OctoPrint instances. That's working from the assumption a user want's to select/deselect the plugins for updating though same as the single button. 

If it's just showing them, I could probably do a little list with the printers next to them. 
For Example. 
Plugin A - OP 1, OP 2
Plugin B - OP 2, OP 3

Should be able to work that into V1.2 rather easily. The selection on them is a little more involved due to some pop up libraries I'd used and would have to work around.

**Reply by taker218** [reply]
> If it's just showing them, I could probably do a little list with the printers next to them.
> For Example.
> Plugin A - OP 1, OP 2
> Plugin B - OP 2, OP 3

That would be totally fine for me, it's just that I don't really want to have a look at all the printers one by one to see the plugins. It would be nicer to have them listed in one window and then update them all at once.

If I would need to exclude a plugin from the updates I would do that via the printer dialogue.

**Reply by NotExpectedYet** [reply]
> > If it's just showing them, I could probably do a little list with the printers next to them.
> > For Example.
> > Plugin A - OP 1, OP 2
> > Plugin B - OP 2, OP 3
> 
> That would be totally fine for me, it's just that I don't really want to have a look at all the printers one by one to see the plugins. It would be nicer to have them listed in one window and then update them all at once.
> 
> If I would need to exclude a plugin from the updates I would do that via the printer dialogue.

I'll convert this to a feature request so I don't forget. 

If I can create some form of unified list I'll try but at the very least I'll display the plugins rather than just a blind update.

**Reply by NotExpectedYet** [reply]
Convert it to an issue for implementing sorry

**Reply by NotExpectedYet** [reply]
#874 for reference :)

**Comment by NotExpectedYet** [comment]
> Hello,
> 
> I'm using OctoFarm for a while now and it would be nice to see, which plugins need to be updated before I'm updating them via OctoFarm. Right now I'm going to the webinterface of my printers and see which ones are due to update.
> 
> It would be nice to see the plugin-names in the confirmation dialogue when updating multiple printers.

![image](https://user-images.githubusercontent.com/61137988/166134353-0ff95265-369d-4561-9844-b4a7b3a56933.png)


Coming in V1.2, will be on development branch today more than likely
